ibis/backends/pyspark/compiler.py

```diff
@@ -223,28 +223,28 @@ def visit_CountDistinctStar(self, op, *, arg, where):
         return self.f.count(sge.Distinct(expressions=cols))
 
     def visit_FirstValue(self, op, *, arg):
-        return self.f.first(arg, TRUE)
+        return self.f.anon.first(arg, TRUE)
 
     def visit_LastValue(self, op, *, arg):
-        return self.f.last(arg, TRUE)
+        return self.f.anon.last(arg, TRUE)
 
     def visit_First(self, op, *, arg, where):
         if where is not None:
             arg = self.if_(where, arg, NULL)
-        return self.f.first(arg, TRUE)
+        return self.f.anon.first(arg, TRUE)
 
     def visit_Last(self, op, *, arg, where):
         if where is not None:
             arg = self.if_(where, arg, NULL)
-        return self.f.last(arg, TRUE)
+        return self.f.anon.last(arg, TRUE)
 
     def visit_Arbitrary(self, op, *, arg, how, where):
         if where is not None:
             arg = self.if_(where, arg, NULL)
         if how == "first":
-            return self.f.first(arg, TRUE)
+            return self.f.anon.first(arg, TRUE)
         elif how == "last":
-            return self.f.last(arg, TRUE)
+            return self.f.anon.last(arg, TRUE)
```

Comment:
Is this `sqlglot` distinguishing between when it "knows" what a given function is vs. when it just takes user input and assumes it maps to something in the target backend?

This seems ripe for future breakage if functions get "promoted" out of the anon space

Reply:
I'm not sure what the answer is to your question, but it appears to be parsing to something that isn't then generated into the exact equivalent (but in theory semantically equivalent):

```
In [1]: import sqlglot as sg, sqlglot.expressions as sge

In [2]: sg.parse_one("select first(x, true)", read="spark")
Out[2]:
Select(
  expressions=[
    IgnoreNulls(
      this=First(
        this=Column(
          this=Identifier(this=x, quoted=False))))])

In [3]: sg.parse_one("select first(x, true)", read="spark").sql('spark')
Out[3]: 'SELECT FIRST(x) IGNORE NULLS'
```

Reply:
sqlglot has _some_ functions that it enforces the semantics of. In this case, I should probably not use `anon` and I should do the proper thing and wrap the unary `first`/`last` call in an `IgnoreNulls` object. I'll do that now.